dataRetriever/dataPool/proofsCache/proofsPool.go

```diff
@@ -0,0 +1,103 @@
+package proofscache
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/multiversx/mx-chain-core-go/data"
+	logger "github.com/multiversx/mx-chain-logger-go"
+)
+
+var log = logger.GetOrCreate("dataRetriever/proofscache")
+
+type proofsPool struct {
+	mutCache sync.RWMutex
+	cache    map[uint32]*proofsCache
+}
+
+// NewProofsPool creates a new proofs pool component
+func NewProofsPool() *proofsPool {
+	return &proofsPool{
+		cache: make(map[uint32]*proofsCache),
+	}
+}
+
+// AddProof will add the provided proof to the pool
+func (pp *proofsPool) AddProof(
+	headerProof data.HeaderProofHandler,
+) error {
+	if headerProof == nil {
+		return ErrNilProof
+	}
+
+	shardID := headerProof.GetHeaderShardId()
+	headerHash := headerProof.GetHeaderHash()
+
+	hasProof := pp.HasProof(shardID, headerProof.GetHeaderHash())
+	if hasProof {
+		log.Debug("there was already an valid proof for header, headerHash: %s", headerHash)
```

Comment:
```suggestion
		log.Debug("there was already a valid proof for header, headerHash: %s", headerHash)
```

and maybe log.Trace.. I think this will happen a lot, when multiple nodes send proofs in paralel